AutomationHelpers/src/UserCodeCollections/WebLibrary.cs

```diff
@@ -89,5 +91,50 @@ public static string GetHttpStatusCode(string url)
 
             return ((int)resp.StatusCode).ToString();
         }
+
+        /// <summary>
+        /// Takes screenshot of entire web page and reports it.
+        /// </summary>
+        /// <param name="repoItemInfo">Repository item</param>
+        [UserCodeMethod]
+        public static void ReportFullPageScreenshot(RepoItemInfo repoItemInfo)
+        {
+            try
+            {
+                if (repoItemInfo == null)
+                {
+                    Report.Error("Repository item must be provided.");
+                    return;
+                }
+
+                WebDocument webDocument = repoItemInfo.CreateAdapter<WebDocument>(false);
+                if (webDocument == null)
+                {
+                    Report.Error("Repository item '" + repoItemInfo.FullName + "' is not web document. " +
+                                 "Screenshot can be taken only for web documents.");
+                    return;
+                }
+
+                var screenshotFilePath = "tmp_screenshot.png";
+
+                var screenshot = webDocument.CaptureFullPageScreenshot();
+                screenshot.Save(screenshotFilePath);
+
+                Report.LogData(ReportLevel.Info, "ScreenShot", screenshot);
+
+                if (File.Exists(screenshotFilePath))
+                {
+                    try
+                    {
+                        File.Delete(screenshotFilePath);
+                    }
+                    catch { }
+                }
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
```

Comment:
This is used very frequently throughout the automation helpers, but is bad practice since you only get the message of the outermost exception. Probably refactor all Report.Error(ex.Message) calls to something more reasonable in a separate commit?
Easiest would be to just use "ex.ToString()", everything else might be too much work. Best would be something like that:
`Report.Log(ReportElvel.Error, "WebLibrary", ex.FullMessage, new SimpleReportMetadata("stacktrace", ex.StackTrace));`
where FullMessage would be an extension method doing something similar like Ranorex.Libs.ExceptionHelper.GetExceptionFullMessage

Reply:
Done. I hope. :)

Reply:
Nice

Reply:
There are still a few ones missing:
FileLibrary:46
FileLibrary:114
SystemLibrary:40

Reply:
Regarding FileLibrary:46. Should I replace Console.WriteLine with Utils.ReportException?

Reply:
IMHO yes, only then it would work with all other Ranorex features like reports on agent.
The automation helpers are a mess, too many (non-dev) people used to push updates to it with no clear guidelines. IMHO there should not be any Console.Write... calls

Reply:
Done